eng/common/pipelines/templates/steps/validate-all-packages.yml

```diff
@@ -10,26 +19,49 @@ steps:
       displayName: "Set as release build"
       condition: and(succeeded(), eq(variables['SetAsReleaseBuild'], ''))
 
-    - task: AzureCLI@2
-      inputs:
-        azureSubscription: opensource-api-connection
-        scriptType: pscore
-        scriptLocation: inlineScript
-        inlineScript: |
-          $(Build.SourcesDirectory)/eng/common/scripts/Validate-All-Packages.ps1 `
-            -ArtifactList ('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json | Select-Object Name) `
-            -ArtifactPath ${{ parameters.ArtifactPath }} `
-            -RepoRoot $(Build.SourcesDirectory) `
-            -APIKey $(azuresdk-apiview-apikey) `
-            -ConfigFileDir '${{ parameters.ConfigFileDir }}' `
-            -BuildDefinition $(System.CollectionUri)$(System.TeamProject)/_build?definitionId=$(System.DefinitionId) `
-            -PipelineUrl $(System.CollectionUri)$(System.TeamProject)/_build/results?buildId=$(Build.BuildId) `
-            -IsReleaseBuild $$(SetAsReleaseBuild)
-        workingDirectory: $(Pipeline.Workspace)
-      displayName: Validate packages and update work items
-      continueOnError: true
-      condition: >-
-        and(
-          succeededOrFailed(),
-          not(endsWith(variables['Build.Repository.Name'], '-pr'))
-        )
+    - ${{ if eq(length(parameters.PackageInfoFiles), 0) }}:
+      - task: AzureCLI@2
+        inputs:
+          azureSubscription: opensource-api-connection
+          scriptType: pscore
+          scriptLocation: inlineScript
+          inlineScript: |
+            $(Build.SourcesDirectory)/eng/common/scripts/Validate-All-Packages.ps1 `
+              -ArtifactList ('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json | Select-Object Name) `
```

Comment:
Let's take and convert this ArtifactList to a set of PackageInfoFiles and eliminate the need to bifurcate the scripts.